hpcflow/sdk/core/workflow.py

```diff
@@ -676,13 +676,15 @@ def add_task(self, task: Task, new_index=None):
         with self.batch_update():
             self._add_task(task, new_index=new_index, parent_events=[])
 
-    def add_task_after(self, task_ref):
-        # TODO: find position of new task, then call add_task
+    def add_task_after(self, task: Task, task_ref=None):
+        task_ref = task_ref or self.tasks[-1]
+        self.add_task(task, new_index=task_ref.index + 1)
         # TODO: add new downstream elements?
         pass
 
-    def add_task_before(self, task_ref):
-        # TODO: find position of new task, then call add_task
+    def add_task_before(self, task: Task, task_ref=None):
+        task_ref = task_ref or self.tasks[0]
```

Comment:
This will also raise an `IndexError` if there are no tasks in the workflow. In this case, I think it would be better to set `new_index = 0` if `task_ref` is `None`.

Reply:
Resolved in [e72c55b](https://github.com/hpcflow/hpcflow-new/pull/348/commits/e72c55bd53d988c530ef5f5b0a2549d947b2dc47).